### PR TITLE
[search] Renamed Query to Processor.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by mapsmetest** [comment]
Build started sha1 is merged.

File: search/processor_factory.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+namespace search
+{
+class SearchProcessorFactory
+{
+public:
+  virtual ~SearchProcessorFactory() = default;
+
+  virtual unique_ptr<Processor> BuildSearchQuery(Index & index, CategoriesHolder const & categories,
```
Comment (ygorshenin) [inline search/processor_factory.hpp]: BuildProcessor

Comment (ygorshenin) [inline search/processor_factory.hpp]: Or just Build() :)

Comment (mpimenov) [inline search/processor_factory.hpp]: done

**Comment by mapsmetest** [comment]
GitHub Watchdog Success

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by mapsmetest** [comment]
Build started sha1 is merged.

**Comment by mapsmetest** [comment]
GitHub Watchdog Success

**Comment by mpimenov** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

File: search/processor_factory.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#include "std/unique_ptr.hpp"
+
+namespace storage
+{
+class CountryInfoGetter;
+}
+
+namespace search
+{
+class SearchProcessorFactory
```
Comment (ygorshenin) [inline search/processor_factory.hpp]: s/SearchProcessorFactory/ProcessorFactory/g

File: search/processor_factory.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{
+public:
+  virtual ~SearchProcessorFactory() = default;
+
+  virtual unique_ptr<Processor> BuildProcessor(Index & index, CategoriesHolder const & categories,
```
Comment (ygorshenin) [inline search/processor_factory.hpp]: Maybe s/BuildProcessor/Build/ ?

File: search/processor_factory.hpp (hunk 2 above)
Comment (mpimenov) [inline search/processor_factory.hpp]: done

File: search/processor_factory.hpp (hunk 3 above)
Comment (mpimenov) [inline search/processor_factory.hpp]: done

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**